### PR TITLE
Switch to using ActerAvatar

### PR DESCRIPTION
**Pull request by gnunicorn**
the cool new widget for all our avatar needs.

### CODE REVIEW

File: app/lib/features/chat/pages/room_page.dart [hunk 1]
```diff
@@ -118,18 +118,19 @@ class _RoomPageState extends State<RoomPage> {
...
-          avatar: roomController.getUserAvatar(userId),
+        child: ActerAvatar(
+          mode: DisplayMode.User,
+          uniqueId: userId,
+          avatarProviderFuture: avtr != null ? remapForAvatar(avtr) : null,
           displayName: roomController.getUserName(userId),
-          radius: 50,
-          isGroup: false,
-          stringName: simplifyUserId(userId)!,
+          size: 50,
```
Comment (gtalha07) [inline app/lib/features/chat/pages/room_page.dart:133]: Is the cache size provided in ```ActerAvatar``` or we are fetching images from rust via thumbnails.

File: app/packages/rust_sdk/lib/acter_flutter_sdk.dart [hunk 2]
```diff
@@ -64,6 +64,18 @@ Color convertColor(ffi.EfkColor? primary, Color fallback) {
   );
 }
 
+Future<ImageProvider<Object>?> remapForAvatar(
+  Future<ffi.FfiBufferUint8?> fut,
+) async {
+  try {
+    final buffered = (await fut)!.asTypedList();
+    return MemoryImage(buffered);
```
Comment (gtalha07) [inline app/packages/rust_sdk/lib/acter_flutter_sdk.dart:72]: Cache size might be needed here?

**Review by gtalha07 — APPROVED** [review]
Looks good. I see you've made it as standalone package, nice. Wondering if you could add test cases for the ```ActerAvatar``` to ensure they work as intended?

**Comment by gnunicorn** [comment]
> Looks good. I see you've made it as standalone package, nice. Wondering if you could add test cases for the `ActerAvatar` to ensure they work as intended?


I was asking about exactly that. The example in the package shows it working nicely, but I don't really understand how to test widgets in unit tests...

**Comment by gnunicorn** [comment]
@gtalha07 applied your suggestions and a few other fixes (in particular in the package itself). //cc @kokodavid this also contains the `DisplayMode.Space` for squared-icons already :) .

grafik
grafik

**Review by gtalha07 — APPROVED** [review]

**Comment by gtalha07** [comment]
> > Looks good. I see you've made it as standalone package, nice. Wondering if you could add test cases for the `ActerAvatar` to ensure they work as intended?
> 
> I was asking about exactly that. The example in the package shows it working nicely, but I don't really understand how to test widgets in unit tests...

I can add those later, approved.

**Pull request merged**